examples/confluent_cloud_example/confluent_cloud_example.go

```diff
@@ -90,14 +123,59 @@ func main() {
 	}
 
 	topics := []string{topic}
-	c.SubscribeTopics(topics, nil)
+	consumer.SubscribeTopics(topics, nil)
 
 	for {
-		msg, err := c.ReadMessage(100 * time.Millisecond)
+		message, err := consumer.ReadMessage(100 * time.Millisecond)
 		if err == nil {
-			fmt.Printf("consumed: %s: %s\n", msg.TopicPartition, string(msg.Value))
+			fmt.Printf("consumed from topic %s [%d] at offset %v: "+
+				string(message.Value), *message.TopicPartition.Topic,
+				message.TopicPartition.Partition, message.TopicPartition.Offset)
 		}
 	}
 
-	c.Close()
+	consumer.Close()
+
+}
+
+func createTopic(topic string) {
+
+	adminClient, err := kafka.NewAdminClient(&kafka.ConfigMap{
+		"bootstrap.servers":       bootstrapServers,
+		"broker.version.fallback": "0.10.0.0",
+		"api.version.fallback.ms": 0,
+		"sasl.mechanisms":         "PLAIN",
+		"security.protocol":       "SASL_SSL",
+		"sasl.username":           ccloudAPIKey,
+		"sasl.password":           ccloudAPISecret})
+
+	if err != nil {
+		fmt.Printf("Failed to create Admin client: %s\n", err)
+		os.Exit(1)
+	}
+	// Contexts are used to abort or limit the amount of time
+	// the Admin call blocks waiting for a result.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Create topics on cluster.
+	// Set Admin options to wait for the operation to finish (or at most 60s)
+	maxDuration, err := time.ParseDuration("60s")
+	if err != nil {
+		panic("ParseDuration(60s)")
+	}
+	results, err := adminClient.CreateTopics(ctx,
+		[]kafka.TopicSpecification{{
+			Topic:             topic,
+			NumPartitions:     1,
+			ReplicationFactor: 3}},
+		kafka.SetAdminOperationTimeout(maxDuration))
+	if err != nil {
```

Comment:
This error only catches request-level errors, not per-topic errors.
You will need to check the status of the topic in `results` to get per-topic creation errors - the reason is that a CreateTopics() call may succeed for some topics and fail for others.
So something like:
`if results[0].Error.Code() != nil { .. topic creation failed for results[0].Topic }`

Reply:
Done. I wrote the following check:

`	// Check for specific topic errors
	for _, result := range results {
		if result.Error.Code() != kafka.ErrNoError {
			fmt.Printf("Topic creation failed for %s: %v",
				result.Topic, result.Error.String())
		}
	}
`